static/docs/user-guide/contributing.md

```diff
@@ -112,6 +112,127 @@ To pass additional arguments:
     $ python -m tests --pdb
 ```
 
+## Testing remotes
+
+This tests require additional effort to set up, so they are skipped by default.
+You don't need this in most cases, however, if you develop or fix some remote
+related code you might need to go through steps below.
+
+Install requirements for whatever remote(s) you are going to test:
+
+```bash
+    pip install -e .[s3]
+    pip install -e .[gs]
+    pip install -e .[azure]
+    pip install -e .[ssh]
+    # or
+    pip install -e .[all]
+```
+
+You will need to update your `ENV` throughout subsequent steps, so we created
+a template for you:
+
+```bash
+    cp remotes_env.sample remotes_env
+```
+
+Then set up lines you need and comment out the rest, the details are explained
+in remote specific subsections. To activate these env vars use:
+
+```bash
+    source remotes_env
+```
+
+If some member of your team had already went through all of this you may just
+ask his `remotes_env` file and Google Cloud credentials and you are all set.
+
+### S3
```

Comment:
let's use <details> to hide remote specific details into expandable sections. Check this doc for example: dvc.org/static/docs/get-started/install.md

Reply:
Not sure I understand this correctly. Do you want to hide everything into single expandable section? Or you want several section, then what do you want to hide? This is all remote specific.

Reply:
I meant something like this https://dvc.org/doc/commands-reference/remote-add#options . One expandable section per remote type, in this case probably one section per ### title.

Reply:
I see.